### DISCUSSION TITLE
Equivalent of System.Drawing.Bitmap.LockBits() in ImageSharp

### DISCUSSION THREAD

**Discussion by kikootwo** [SixLabors/ImageSharp#1264] [Q&A] [answered]
I am working on fully replacing System.Drawing.Bitmap with ImageSharp in my solution. Is there an equivalent Bitmap.LockBits() in ImageSharp? I see some discussion from 2018, but I assume things have changed since then.

Thanks in advance for your help!

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah... 2018 is a long time ago in development years 😉

The pixel buffer in ImageSharp is laid out in memory in the manner dictated by the `TPixel` generic parameter which allows fast lock free access to the buffer.

https://docs.sixlabors.com/articles/imagesharp/pixelbuffers.html

**Reply by kikootwo** [reply]
Thank you so much for your quick response! That looks like exactly what I'm after. I am instantiating an 'Image' with Image.Load(path) but my image doesn't have that 'TryGetPixelSpan' Is there a namespace/other package I'm missing?

**Reply by kikootwo** [reply]
Nevermind, please ignore me!

For aynone else reading this, your image must have a pixel format for example` Image<Rbga>`

Thank you again!

**Reply by JimBobSquarePants** [reply]
Yep, that’s right. If you use the generic version you get more options
